src/services/webhook.js

```diff
@@ -298,14 +303,48 @@ class WebhookController {
             .catch(err => {
                 if (err) {
                     if (retryCount < this.retryCount) {
-                        console.error('[WebhookController] Error occurred, trying again:', err);
+                        console.error(`[WebhookController] Error occurred, trying again for ${url}`);
                         this.sendEvents(events, url, retryCount++);
                     } else {
-                        console.error('[WebhookController] Error occurred, max retry count reached:', err);
+                        console.error(`[WebhookController] Error occurred, max retry count reached for ${url}`);
                     }
                 }
-            });
+          });
     }
+
+    async checkOnline() {
+        let prevReq = {}
+        await Promise.all(this.urls.map(url => {
+            if (prevReq[url]) {
+                prevReq[url].cancel()
+            }
+            prevReq[url] = axios.CancelToken.source()
+            axios({
```

Comment:
You did not return the Promise, nor did you async/await.

Reply:
🤔 But he .then and .catch's to enforce the await, so should be fine imo?

Reply:
`then` and `catch` returns another Promise I think.

Reply:
🤔 I think he's good tbh but I'm no JS dev https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then